custom_menu_template.gs

```diff
@@ -0,0 +1,57 @@
+/**
+* A custom menu template for the following - 
+* Simple Custom Menu
+* Custom Menu with separators
+* Custom Menu with sub-menus
+* Adding an item to the editor's Add-on menu
+*
+* The onOpen event handler is triggered when the attached spreadsheet is opened.
+* Class SpreadsheetApp is used to access and create Google Sheets files.
+*/
+function onOpen() {
+  // The getUi method returns an instance of the spreadsheet's UI environment.
+  // It allows the script to add features like menus, dialogs, sidebars.
+  var spreadsheetUi = SpreadsheetApp.getUi(); 
+  
+  // A simple Custom Menu 
+  spreadsheetUi.createMenu('Custom Menu 1')
+      .addItem('First item', 'menuItem1') // Adds an item to the menu.
+      .addItem('Second item', 'menuItem2')
+      .addToUi();  // Inserts the menu into the instance of the editor's user interface.
+ 
+  // Custom Menu with separators
+  spreadsheetUi.createMenu('Custom Menu 2')
+      .addItem('First item', 'menuItem1')
+      .addSeparator() // Adds a visual separator to the menu.
+      .addItem('Second item', 'menuItem2')
+      .addSeparator() 
+      .addItem('Third item', 'menuItem3')
+      .addToUi();  
+  
+  // Custom Menu with a sub-menu
+  spreadsheetUi.createMenu('Custom Menu 3')
+      .addSubMenu(spreadsheetUi.createMenu('Sub-menu') // Adds a sub-menu to the menu.
+          .addItem('First item', 'menuItem1'))
+      .addSeparator() 
+      .addItem('Second item', 'menuItem2')
+      .addSeparator() 
+      .addItem('Third item', 'menuItem3')
+      .addToUi();  
+    
+  // Add an item to the editor's Add-on menu, under a sub-menu whose name is set automatically.
+  spreadsheetUi.createAddonMenu()
+      .addItem('First Item', 'menuItem1')
+      .addToUi();
+}
+
+function menuItem1() {
+  // Add functionality to be performed on clicking the menu item here.
```

Comment:
I meant only remove the implementation from menuItem3() function. You can keep menuItem1 and menuItem2 function implementations as before. Also, mark resolve conversation whenever you resolve a comment. :)

Reply:
Made the correction. 
Noted, I will resolve all of them from now!